src/telempostdaemon.c

```diff
@@ -0,0 +1,632 @@
+                telem_log(LOG_DEBUG, "Processing staged record: %s\n",
+                          namelist[i]->d_name);
+                ret = asprintf(&record_path, "%s/%s", DEFAULT_STAGE_DIR, namelist[i]->d_name);
+                if (ret == -1) {
+                        telem_log(LOG_ERR, "Failed to allocate memory for staging record full path\n");
+                        exit(EXIT_FAILURE);
+                }
+                if (process_staged_record(record_path, daemon)) {
+                        unlink(record_path);
+                }
+                free(record_path);
+        }
+
+        for (int i = 0; i < numentries; i++) {
+                free(namelist[i]);
+        }
+        free(namelist);
+}
+
+void run_daemon(TelemPostDaemon *daemon)
+{
+        int ret;
+        int spool_process_time = spool_process_time_config();
+        bool daemon_recycling_enabled = daemon_recycling_enabled_config();
+        time_t last_spool_run_time = time(NULL);
+        time_t last_daemon_start_time = time(NULL);
+
+        assert(daemon);
+        assert(daemon->pollfds);
+        assert(daemon->pollfds[signlfd].fd);
+        assert(daemon->pollfds[watchfd].fd);
+
+        while (1) {
+                malloc_trim(0);
+                ret = poll(daemon->pollfds, NFDS, spool_process_time * 1000);
+                if (ret == -1) {
+                        telem_perror("Failed to poll daemon file descriptors");
+                        break;
+                } else if (ret != 0) {
+
+                        if (daemon->pollfds[signlfd].revents != 0) {
+                                struct signalfd_siginfo fdsi;
+                                ssize_t s;
+                                s = read(daemon->sfd, &fdsi, sizeof(struct signalfd_siginfo));
+                                if (s != sizeof(struct signalfd_siginfo)) {
+                                        telem_perror("Error while reading from the signal"
+                                                     "file descriptor");
+                                        exit(EXIT_FAILURE);
+                                }
+
+                                if (fdsi.ssi_signo == SIGTERM || fdsi.ssi_signo == SIGINT) {
+                                        telem_log(LOG_INFO, "Received either a \
+                                                                     SIGINT/SIGTERM signal\n");
+                                        break;
+                                }
+                        } else if (daemon->pollfds[watchfd].revents != 0) {
+
+                                int ret = 0;
+                                ssize_t i = 0;
+                                ssize_t length = 0;
+                                char buffer[BUFFER_LEN];
+
+                                length = read(daemon->fd, buffer, BUFFER_LEN);
+                                if (length < 0) {
+                                        perror("Reading descriptor returned error");
```

Comment:
An error message like this is not actionable. Either ignore it or provide info to help debug